addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -0,0 +1,198 @@
+            processed = Picking.browse()
+            unsatisfied_state = lambda p: p.state not in ("assigned", "cancel", "done")
+
+            while reserve_all or to_reserve > 0:
+
+                if self:
+                    pickings = self.filtered(
+                        lambda p: p.picking_type_id == picking_type
+                    )
+                    # Removed processed pickings from self
+                    pickings -= processed
+                else:
+                    domain = base_domain[:]
+                    if processed:
+                        domain.append(("id", "not in", processed.ids))
+                    pickings = Picking.search(domain, limit=limit)
+
+                _logger.info(f"Reserving stock for pickings {pickings}.")
+                if not pickings:
+                    # No pickings left to process.
+                    # If u_num_reservable_pickings is -1, or there are
+                    # fewer available pickings that the limit, the loop must
+                    # terminate here.
+                    break
+
+                batch = pickings.batch_id
+                if batch and batch.state == "draft":
+                    # Add to seen pickings so that we don't try to process
+                    # this batch again.
+                    processed |= batch.picking_ids
+                    continue
+
+                if batch and picking_type.u_reserve_batches:
+                    pickings = batch.picking_ids
+
+                # MPS: mimic Odoo's retry behaviour
+                tries = 0
+                while True:
+
+                    try:
+                        with self.env.cr.savepoint():
+                            pickings = pickings._reserve_stock_assign()
+                            batch._compute_state()
+
+                            processed |= pickings
+
+                            unsatisfied = pickings.filtered(unsatisfied_state)
+
+                            mls = pickings.move_line_ids
+                            if unsatisfied:
+                                # Unreserve if the picking type cannot handle partials or it
+                                # can but there is nothing allocated (no stock.move.lines)
+                                if not mls:
+                                    # construct error message, report only products
+                                    # that are unreservable.
+                                    moves = unsatisfied.move_lines.filtered(
+                                        unsatisfied_state
+                                    )
+                                    moves_name = moves.product_id.name_get()
+                                    products = moves.product_id.default_code
+                                    picks = moves.picking_id.name
+                                    msg = (
+                                        f"Unable to reserve stock for products {products} with record {moves_name} "
```

Comment:
We don't need `moves_name` here.

Just do: `products = moves.product_id.name_get()` and have the error message as it was before. :)